src/main/java/com/redhat/devtools/lsp4ij/usages/LSPUsageTriggeredPsiElement.java

```diff
@@ -39,4 +42,22 @@ public List<LocationData> getLSPReferences() {
     public void setLSPReferences(List<LocationData> references) {
         this.references = references;
     }
+
+    @Nullable
+    public PsiElement getRealElement() {
+        return getContainingFile().findElementAt(getTextOffset());
+    }
+
+    @Override
+    @Nullable
+    public String getLocationString() {
+        // If this is reference, show the location string of the target

```

Comment:
By default the location string is the name of the containing file for the _source_ element from which usages were triggered. If that's a reference to something in another file, the location string is incorrect and misleading. Instead see if the source element is a reference and, if so, show the containing file name of the _target_, e.g.:

![image](https://github.com/user-attachments/assets/01655640-cb11-4ed6-b164-f0ad82733ce3)